src/d3a/d3a_core/sim_results/endpoint_buffer.py

```diff
@@ -146,6 +147,13 @@ def _populate_core_stats_and_sim_state(self, area):
                 core_stats_dict['trades'].append(trade.serializable_dict())
         if hasattr(area.current_market, 'market_fee'):
             core_stats_dict['market_fee'] = area.current_market.market_fee
+        if hasattr(area.current_market, 'const_fee_rate') is not None and \
+                area.current_market is not None:
```

Comment:
```suggestion
        if hasattr(area.current_market, 'const_fee_rate') is not None and \
                area.current_market.const_fee_rate is not None:
```

Reply:
Can you please explain the motive behind the proposed change?

Reply:
In your code you are checking if `hasattr(area.current_market, 'const_fee_rate') is not None`, this is not a valid check, the `hasattr` will return a boolean either True or False, and there is no way it will return a None (The value you're comparing against)
Furthermore, the same check is intended to validate if area.current_market has an attrribute `const_fee_rate`, so the second condition `area.current_market is not None` doesn't make sense at this stage, if there is a possibility that this can be None it should be the first condition

Reply:
```suggestion
        if (getattr(area.current_market, "const_fee_rate", None) is not None and
                area.current_market is not None):
```
How about this? 

`area.current_market is not None` is needed when its called at the start of simulation and `current_market` is None at that point.